File: app/src/main/java/pl/mosenko/sunnypodlaskie/persistence/dao/WeatherDataEntityDAO.java
package pl.mosenko.sunnypodlaskie.persistence.dao;

import com.j256.ormlite.support.ConnectionSource;
import com.j256.ormlite.table.DatabaseTableConfig;

import java.sql.SQLException;

import pl.mosenko.sunnypodlaskie.persistence.entities.WeatherDataEntity;

/**
 * Created by syk on 16.05.17.
 */

public class WeatherDataEntityDAO extends AbstractBaseDaoImpl<WeatherDataEntity, Long> {

    public WeatherDataEntityDAO(Class<WeatherDataEntity> dataClass) throws SQLException {
        super(dataClass);
    }

    public WeatherDataEntityDAO(ConnectionSource connectionSource) throws SQLException {
        super(connectionSource, WeatherDataEntity.class);
    }

    public WeatherDataEntityDAO(ConnectionSource connectionSource, DatabaseTableConfig<WeatherDataEntity> tableConfig) throws SQLException {
        super(connectionSource, tableConfig);
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/persistence/dao/WeatherDataEntityDao.kt
package pl.mosenko.sunnypodlaskie.persistence.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow
import pl.mosenko.sunnypodlaskie.persistence.model.WeatherData

/**
 * Created by syk on 16.05.17.
 */
@Dao
interface WeatherDataEntityDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(weatherDataList: List<WeatherData>)

    @Query("SELECT * FROM weather_data")
    suspend fun getAllWeatherData(): List<WeatherData>?

    @Query("DELETE FROM weather_data")
    suspend fun clearAllWeatherData()

    @Query("SELECT * FROM weather_data where city = :city")
    fun getWeatherDataByCityName(city: String): Flow<WeatherData>
}
